app/assets/javascripts/header_footer.js

```diff
@@ -1,11 +1,19 @@
 jQuery(document).ready(function() {	 
-	var duration = 300;
+	
+
+	$(window).scroll(function() {
+    if ($(this).scrollTop() >= 50) {        // If page is scrolled more than 50px
+        $('.back-to-top').fadeIn(200);    // Fade in the button
```

Comment:
Instead of using fadeIn, you should probably animate the opacity to something like 0.7

Reply:
You can also remove the event listener after the button fades in and add after the scroll to top is complete.